backends/p4tools/modules/testgen/targets/bmv2/test_backend/ptf.cpp

```diff
@@ -91,125 +44,7 @@ std::vector<std::pair<size_t, size_t>> PTF::getIgnoreMasks(const IR::Constant *m
-    rulesJson["optional_matches"] = inja::json::array();
-
-    rulesJson["act_args"] = inja::json::array();
-    rulesJson["needs_priority"] = false;
-
-    // Iterate over the match fields and segregate them.
-    for (const auto &match : matches) {
-        const auto fieldName = match.first;
-        const auto &fieldMatch = match.second;
-
-        inja::json j;
-        j["field_name"] = fieldName;
-        if (const auto *elem = fieldMatch->to<Exact>()) {
-            j["value"] = formatHexExpr(elem->getEvaluatedValue()).c_str();
-            rulesJson["single_exact_matches"].push_back(j);
-        } else if (const auto *elem = fieldMatch->to<Range>()) {
-            j["lo"] = formatHexExpr(elem->getEvaluatedLow()).c_str();
-            j["hi"] = formatHexExpr(elem->getEvaluatedHigh()).c_str();
-            rulesJson["range_matches"].push_back(j);
-        } else if (const auto *elem = fieldMatch->to<Ternary>()) {
-            j["value"] = formatHexExpr(elem->getEvaluatedValue()).c_str();
-            j["mask"] = formatHexExpr(elem->getEvaluatedMask()).c_str();
-            rulesJson["ternary_matches"].push_back(j);
-            // If the rule has a ternary match we need to add the priority.
-            rulesJson["needs_priority"] = true;
-        } else if (const auto *elem = fieldMatch->to<LPM>()) {
-            j["value"] = formatHexExpr(elem->getEvaluatedValue()).c_str();
-            j["prefix_len"] = elem->getEvaluatedPrefixLength()->value.str();
-            rulesJson["lpm_matches"].push_back(j);
-        } else if (const auto *elem = fieldMatch->to<Optional>()) {
-            j["value"] = formatHexExpr(elem->getEvaluatedValue()).c_str();
-            if (elem->addAsExactMatch()) {
-                j["use_exact"] = "True";
-            } else {
-                j["use_exact"] = "False";
-            }
-            rulesJson["needs_priority"] = true;
-            rulesJson["optional_matches"].push_back(j);
-        } else {
-            TESTGEN_UNIMPLEMENTED("Unsupported table key match type \"%1%\"",
-                                  fieldMatch->getObjectName());
-        }
-    }
-
-    for (const auto &actArg : args) {
-        inja::json j;
-        j["param"] = actArg.getActionParamName().c_str();
-        j["value"] = formatHexExpr(actArg.getEvaluatedValue()).c_str();
-        rulesJson["act_args"].push_back(j);
-    }
-
-    return rulesJson;
-}
-
-inja::json PTF::getSend(const TestSpec *testSpec) {
-    const auto *iPacket = testSpec->getIngressPacket();
-    const auto *payload = iPacket->getEvaluatedPayload();
-    inja::json sendJson;
-    sendJson["ig_port"] = iPacket->getPort();
-    auto dataStr = formatHexExpr(payload, false, true, false);
-    sendJson["pkt"] = insertHexSeparators(dataStr);
-    sendJson["pkt_size"] = payload->type->width_bits();
-    return sendJson;
-}
-
-inja::json PTF::getVerify(const TestSpec *testSpec) {
+inja::json PTF::getExpectedPacket(const TestSpec *testSpec) const {
```

Comment:
Again, this looks very similar except for the ignoring masks part and what I assume is minor formatting?

Reply:
`getVerify` was not very descriptive, so I renamed the function.

Reply:
Sorry, I meant that it looks very similar to the 'common' function, making me think they could be combined.

Reply:
There is a subtle difference: `getIgnoreMasks(payloadMask)` is called, which is not the case in the other frameworks. I will think about how to unify this.